src/System.Collections/src/System/Collections/Generic/Comparer.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Diagnostics.Contracts;
+
+namespace System.Collections.Generic
+{
+    public abstract class Comparer<T> : IComparer, IComparer<T>
+    {
+        protected Comparer()
+        {
+        }
+
+        public static Comparer<T> Default
+        {
+            get
+            {
+                if (_default == null)
+                    _default = CreateComparer();
```

Comment:
Probably something to be done as a separate commit, but see https://github.com/dotnet/corefx/issues/780